kubernetes/resource_kubernetes_endpoints_test.go

```diff
@@ -259,10 +267,8 @@ func testAccCheckEndpointSubsets(svc *api.Endpoints, expected []api.EndpointSubs
 		}
 
 		subsets := svc.Subsets
-
-		if !reflect.DeepEqual(subsets, expected) {
-			return fmt.Errorf("Endpoint subsets don't match.\nExpected: %#v\nGiven: %#v",
-				expected, svc.Subsets)
+		if diff := cmp.Diff(expected, subsets); diff != "" {
```

Comment:
I had to replace `reflect.DeepEqual()` usage by `cmd.Diff()` because the former always returns false for values containing non-nil function values which is the case with `schema.Set`.

This resulted in unexpected errors with no visible differences:

```
=== RUN   TestFlattenEndpointsSubsets
--- FAIL: TestFlattenEndpointsSubsets (0.00s)
    structure_endpoints_test.go:180: Unexpected output from flattener.
        Expected: []interface {}{map[string]interface {}{"address":[]interface {}{map[string]interface {}{"hostname":"any.hostname.io", "ip":"10.0.0.4", "node_name":"test-nodename"}}, "not_ready_address":[]interface {}{map[string]interface {}{"hostname":"notready.hostname.io", "ip":"10.0.0.5"}}, "port":*Set(map[string]interface {}{"3203037011":map[string]interface {}{"name":"transport", "port":8889, "protocol":"UDP"}})}}
        Given:    []interface {}{map[string]interface {}{"address":[]interface {}{map[string]interface {}{"hostname":"any.hostname.io", "ip":"10.0.0.4", "node_name":"test-nodename"}}, "not_ready_address":[]interface {}{map[string]interface {}{"hostname":"notready.hostname.io", "ip":"10.0.0.5"}}, "port":*Set(map[string]interface {}{"3203037011":map[string]interface {}{"name":"transport", "port":8889, "protocol":"UDP"}})}}
```